content/articles/searching-1.md

```diff
@@ -0,0 +1,257 @@
+search is good enough. For the mapping project
+_[Writ Large](https://writlarge.ctl.columbia.edu/map/)_,
+a Django project with a manageable size of data set, we use Django PostgreSQL
+search tools. We decided to go with
+[Solr](http://lucene.apache.org/solr/) for
+[Footprints](https://footprints.ccnmtl.columbia.edu/search/) because there are
+thousands of records to go through, using much more robust faceted search.
+
+However, considerations for search doesn’t simply stop with the technology.
+Let’s go back to the “Green Notebook Quest.” The search was a two-step process.
+Initially I did a simple search, then I filtered the results with certain
+parameters.
+
+```
+Initial query "green notebook"
+---------------
+Filter by:
+    size: "small"
+    tag: "cat"
+    use: "math"
+```
+
+But wait, the initial query can still be broken down to a two-step process:
+
+```
+Initial query:
+    q1:
+        type: "notebook"
+    q2:
+        color: "green"
+---------------
+Filter by:
+    size: "small"
+    tag: "cat"
+    use: "math"
+```
+
+So now, we have to decide how to design the UI and UX around this simple search
+pattern. Can we assume that the user has a certain knowledge of superuser
+searches so we can condense everything into one search field, i.e.
+`type:notebook AND color:green AND size:"small"` etc? Or should we design an
+elegant search interface that appeals more to the user’s intuition? We did the
+latter for [Footprints](https://footprints.ccnmtl.columbia.edu/search/) search.
+
+### Where is the user searching? ###
+The may be instances where we set limits to user searches (search only
+published or public records if user is not an admin, for example). What fields
+or attributes are being exposed to the user search? The users may want to set
+their own limits to the search as well. The latter case can also be thought as
+a facet of the search. The areas of search in “Green Notebook Quest,” for
+example, were “shelves by the cabinet in my room”
+
+Why is this important? As always, for UI, and UX.
+
+### Who can see what the user is looking for? ###
+Sometimes, we show to hide certain search results based on user permissions.
+Only editors will see unpublished articles, for example. This type of decisions
+needs to be included in testing so we can be certain that a zero match is
+indeed and intended match.
+
+### How can the user see what the user looking for? ###
+We ask this question so we can design the search result page. Should the
+results page be separate? For
+[Footprints](https://footprints.ccnmtl.columbia.edu/search/), we decided to
+show all records at first, and then show only records that match the search
+queries. This blog’s simple search displays mathing posts in a pop-up panel. In
+_[Writ Large](https://writlarge.ctl.columbia.edu/map/)_, all pins that don’t
+match the search criteria are dimmed instead of hidden.
+
+### The result from the user’s search, does it match what the user is looking
```

Comment:
This line is not displaying properly in the preview. Possibly a CRLF after `looking`?